### PR TITLE
Java: Modularize wrapper

### PR DESCRIPTION
**Pull request by jonathanl-bq**
This change exposes the API package only using Java's new module system. It also updates the name of our JAR from "client" to "glide-client".

This requires the upgrading protobuf to the 4.26.0-rc2 release candidate, which fixes an issue where protobuf was not a modularized project (see https://github.com/protocolbuffers/protobuf/issues/12639).

### CODE REVIEW

File: java/client/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ repositories {
 }
 
 dependencies {
-    implementation group: 'com.google.protobuf', name: 'protobuf-java', version: '3.24.3'
+    implementation group: 'com.google.protobuf', name: 'protobuf-java', version: '4.26.0-RC2'
```
Comment (Yury-Fridlyand) [inline java/client/build.gradle:12]: Is it compatible with protobuf we install CI?

Comment (Yury-Fridlyand) [reply java/client/build.gradle]: No it is not...

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — COMMENTED** [review]
Should we update the README? 
We should indicate that they need to install protoc >= 26.1, right?

**Comment by Yury-Fridlyand** [comment]
Why it triggers so much unrelated GHAs? npm-cd?

**Comment by jonathanl-bq** [comment]
> Why it triggers so much unrelated GHAs? npm-cd?

Yeah, I'm confused by this too. My change was okay when I updated protoc just for the Java workflow. But putting it in shared dependencies seems to be causing a lot of weird failures. I'm still looking into this, but would appreciate any help.

**Review by acarbonetto — APPROVED** [review]
Approval conditional that we check protoc version in the gradle build.

File: java/client/build.gradle [hunk 2]
```diff
@@ -31,8 +31,29 @@ dependencies {
     testImplementation group: 'org.mockito', name: 'mockito-junit-jupiter', version: '3.12.4'
 }
 
+ext {
+    checkProtocVersion = { String output ->
+        // Line in format like: libprotoc 26.1
+        int majorVersion = java.lang.Integer.parseInt(output.split(" ")[1].split("\\.")[0].trim());
+        int minorVersion = java.lang.Integer.parseInt(output.split(" ")[1].split("\\.")[1].trim());
+        if (majorVersion < 26 || (majorVersion == 26 && minorVersion < 1)) {
```
Comment (Yury-Fridlyand) [inline java/client/build.gradle:39]: Is it possible to get numbers from the dependency instead of hardcoded ones?

**Review by Yury-Fridlyand — APPROVED** [review]
I see odd CI didn't start, perhaps that was a glitch or so

File: java/client/build.gradle [hunk 3]
```diff
@@ -31,8 +31,29 @@ dependencies {
     testImplementation group: 'org.mockito', name: 'mockito-junit-jupiter', version: '3.12.4'
 }
 
+ext {
+    checkProtocVersion = { String output ->
+        // Line in format like: libprotoc 26.1
+        int majorVersion = java.lang.Integer.parseInt(output.split(" ")[1].split("\\.")[0].trim());
+        int minorVersion = java.lang.Integer.parseInt(output.split(" ")[1].split("\\.")[1].trim());
```
Comment (Yury-Fridlyand) [inline java/client/build.gradle:38]:
```suggestion
        int majorVersion = Integer.parseInt(output.split(" ")[1].split("\\.")[0].trim());
        int minorVersion = Integer.parseInt(output.split(" ")[1].split("\\.")[1].trim());
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**